### PR TITLE
[flutter_tools] add EACCES to list of immediate exit tool conditions

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Similar to the permission denied error on Windows, this is not resolvable by the tool.

### CODE REVIEW

File: packages/flutter_tools/lib/src/base/error_handling_file_system.dart [hunk 1]
```diff
@@ -373,7 +373,7 @@ T _runSync<T>(T Function() op, {
   } on FileSystemException catch (e) {
     if (platform.isWindows) {
       _handleWindowsException(e, failureMessage);
-    } else if (platform.isLinux) {
+    } else if (platform.isLinux || platform.isMacOS) {
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/base/error_handling_file_system.dart:376]: seems like we could rename this to posix exception, et cetera

File: packages/flutter_tools/test/general.shard/base/error_handling_io_test.dart [hunk 2]
```diff
@@ -194,6 +194,7 @@ void main() {
 
   group('throws ToolExit on Linux', () {
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/base/error_handling_io_test.dart:195]: on Linux/macOS

File: packages/flutter_tools/lib/src/base/error_handling_io.dart [hunk 3]
```diff
@@ -382,14 +382,14 @@ T _runSync<T>(T Function() op, {
     if (platform.isWindows) {
       _handleWindowsException(e, failureMessage, e.osError?.errorCode ?? 0);
     } else if (platform.isLinux) {
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/base/error_handling_io.dart:384]: ` || platform.isMacOS`?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/base/error_handling_io.dart:384]: oops, messed up my merge

File: packages/flutter_tools/lib/src/base/error_handling_io.dart [hunk 4]
```diff
@@ -490,11 +490,12 @@ class ErrorHandlingProcessManager extends ProcessManager {
   }
 }
 
-void _handleLinuxException(Exception e, String message, int errorCode) {
+void _handlePosixException(Exception e, String message, int errorCode) {
   // From:
   // https://github.com/torvalds/linux/blob/master/include/uapi/asm-generic/errno.h
   // https://github.com/torvalds/linux/blob/master/include/uapi/asm-generic/errno-base.h
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/base/error_handling_io.dart:496]: nit. is there either POSIX documentation for these, or macos specific docs?

**Review by christopherfujino — APPROVED** [review]
LGTM with a nit about updating comment

File: packages/flutter_tools/lib/src/base/error_handling_io.dart (hunk 4 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/base/error_handling_io.dart:496]: done

**Comment by jonahwilliams** [comment]
google testing failure is due to bad bot merge, fyi @renyou

**Pull request merged**